extensions/jms/src/main/java/org/mule/extensions/jms/api/source/JmsSubscribe.java

```diff
@@ -0,0 +1,303 @@
+import static org.mule.extensions.jms.internal.common.JmsOperationCommons.resolveMessageContentType;
+import static org.mule.extensions.jms.internal.common.JmsOperationCommons.resolveOverride;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.slf4j.LoggerFactory.getLogger;
+import org.mule.extensions.jms.api.config.AckMode;
+import org.mule.extensions.jms.api.config.JmsConfig;
+import org.mule.extensions.jms.api.config.JmsConsumerConfig;
+import org.mule.extensions.jms.api.connection.JmsConnection;
+import org.mule.extensions.jms.api.connection.JmsSession;
+import org.mule.extensions.jms.api.destination.ConsumerType;
+import org.mule.extensions.jms.api.exception.JmsExtensionException;
+import org.mule.extensions.jms.api.message.JmsAttributes;
+import org.mule.extensions.jms.api.message.MessageBuilder;
+import org.mule.extensions.jms.internal.consume.JmsMessageConsumer;
+import org.mule.extensions.jms.internal.message.JmsResultFactory;
+import org.mule.extensions.jms.internal.metadata.JmsOutputResolver;
+import org.mule.extensions.jms.internal.publish.JmsPublishParameters;
+import org.mule.extensions.jms.internal.support.Jms102bSupport;
+import org.mule.extensions.jms.internal.support.JmsSupport;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.message.Error;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.scheduler.SchedulerService;
+import org.mule.runtime.core.util.StringMessageUtils;
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.extension.api.annotation.dsl.xml.XmlHints;
+import org.mule.runtime.extension.api.annotation.execution.OnError;
+import org.mule.runtime.extension.api.annotation.execution.OnSuccess;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataScope;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.NullSafe;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.UseConfig;
+import org.mule.runtime.extension.api.annotation.param.display.Summary;
+import org.mule.runtime.extension.api.annotation.source.EmitsResponse;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.source.Source;
+import org.mule.runtime.extension.api.runtime.source.SourceCallback;
+import org.mule.runtime.extension.api.runtime.source.SourceCallbackContext;
+
+import javax.inject.Inject;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.Queue;
+import javax.jms.Topic;
+
+import org.slf4j.Logger;
+
+/**
+ * JMS Subscriber for {@link Destination}s, allows to listen
+ * for incoming {@link Message}s
+ *
+ * @since 4.0
+ */
+@Alias("subscribe")
```

Comment:
source names must be nouns. Isn't listener a descriptive name which is also consistent with http?